frontera/contrib/scrapy/schedulers/frontier.py

```diff
@@ -124,11 +134,13 @@ def process_exception(self, request, exception, spider):
     def open(self, spider):
         log.msg('Starting frontier', log.INFO)
         if not self.frontier.manager.auto_start:
-            self.frontier.start()
+            scrapy_kwargs = {'spider': spider}
+            self.frontier.start(**scrapy_kwargs)
```

Comment:
@kalessin It's quite complex, and overall system design isn't clear just from code. I noticed there spider callbacks:
- ``get_producer_slot(request)``
- ``cf_make_request(fingerprint, qdata, request_cls)``
however, it's still not clear why are they there and how they used.